### PR TITLE
Drafted README.md and CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by matthew-carroll**
Drafted README.md and CONTRIBUTING.md

### CODE REVIEW

**Review by salihgueler — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,3 +1,60 @@
...
+
+## Project principles
+
+`super_editor` is intended to be used by an unbounded variety of apps for editing and rendering documents. Therefore, `super_editor` is built with certain principles in mind. These principles are to be maintained throughout the development process.
+
+**Principles**
+
+* Aggressive composition: Build small, effective tools that are designed to work with other small tools to achieve broader functionality.
+* Establish strong encapsulation boundaries: Keep unrelated concepts completely independent (including transitive dependencies).
+* If you can't see it, it doesn't exist: Every feature should have a runnable demo.
```
Comment (bk-one) [inline CONTRIBUTING.md:13]: love it :)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,3 +1,60 @@
...
+* Tests are not optional: Every feature must be tested, and those tests must run in CI.
+
+**A few specifics**
+
+* Classes and behaviors that are required for `super_editor` but are not specific to the project goals should be placed within the `infrastructure` package.
+* The `core` package should only include the project's fundamental abstractions. Nothing in the `core` package should depend on any code outside the `core` package, except (rarely) something in the `infrastructure` package.
+* Meaningless names like "manager", "helper", "utility", "data", "model" will not be accepted unless there is a strong contextual reason to do so.
+* If the term "controller" is used, it must specifically refer to the Flutter concept of a controller.
```
Comment (bk-one) [inline CONTRIBUTING.md:21]: should we also add that everything needs to be Dart/Flutter only? Not sure if this isn't clear by itself - but just to be explicit about it - so no-one is adding/using platform-specific code

File: README.md [hunk 3]
```diff
@@ -1,9 +1,86 @@
...
-## Spikes
+Super Editor is an open source, configurable, extensible text editor and document renderer for Flutter apps.
 
-Develop spikes within the `/example` project. Include a new entrypoint for each
-spike so that the standard demo continues to work.
+`super_editor` is supported by [Superlist](https://superlistapp.com).
 
-Each spike should include internal comment documentation about which questions
-the spike attempts to answer and what conclusions the spike demonstrates.
+`super_editor` was implemented, and is led by [SuperDeclarative!](https://superdeclarative.com).
```
Comment (bk-one) [inline README.md:7]: I think I'd rather have superdeclarative and superlist on the same level. I'll also have more devs looking into contributing now that we have a first level. So I'd rather go with:

`super_editor` was initiated by Superlist and is being implemented and maintained by SuperDeclarative!, Superlist, and the contributors.

or similar.

File: CONTRIBUTING.md (hunk 2 above)
Comment (matthew-carroll) [reply CONTRIBUTING.md:21]: Sure. Updated.

File: README.md (hunk 3 above)
Comment (matthew-carroll) [reply README.md]: Changed to the suggested statement.

**Pull request merged**